File: GameFramePro/Assets/Scripts/Core/Base/Module/NetWork/Download/Downloader/UnityWebRequest/TextureDownloadManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;
using UnityEngine.Networking;

namespace GameFramePro.NetWorkEx
{
    /// <summary>
    /// 下载图片
    /// </summary>
    public class TextureDownloadManager : UnityWebRequestDownLoadManager
    {
        protected static TextureDownloadManager s_Instance = null;
        public static TextureDownloadManager S_Instance
        {
            get
            {
                if (s_Instance == null)
                    s_Instance = new TextureDownloadManager();
                return s_Instance;
            }
        }// 



        protected override UnityWebRequestDownloadTask GetDownloadTaskInstance(string taskUrl, Action<UnityWebRequest, bool, string> callback, UnityTaskPriorityEnum priorityEnum, params object[] otherParameter)
        {
            var newTask = mUnityWebRequestTaskPoolManager.GetItemFromPool();
            UnityWebRequest webRequest = new UnityWebRequest(taskUrl);
            webRequest.downloadHandler = new DownloadHandlerTexture();
            newTask.InitialedDownloadTask(taskUrl, webRequest, priorityEnum, callback);
            return newTask;
        }
    }
}

File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/NetWorkEx/TextureDownloadManager.lua
---@class CS.GameFramePro.NetWorkEx.TextureDownloadManager : CS.GameFramePro.NetWorkEx.UnityWebRequestDownLoadManager
CS.GameFramePro.NetWorkEx.TextureDownloadManager = {}

---@property readonly CS.GameFramePro.NetWorkEx.TextureDownloadManager.S_Instance : CS.GameFramePro.NetWorkEx.TextureDownloadManager
CS.GameFramePro.NetWorkEx.TextureDownloadManager.S_Instance = nil

---@return CS.GameFramePro.NetWorkEx.TextureDownloadManager
function CS.GameFramePro.NetWorkEx.TextureDownloadManager()
end

---@param taskUrl : CS.System.String
---@param callback : CS.System.Action
---@param priorityEnum : CS.GameFramePro.NetWorkEx.UnityTaskPriorityEnum
function CS.GameFramePro.NetWorkEx.TextureDownloadManager:GetDataFromUrl(taskUrl, callback, priorityEnum)
end